src/codeu/chat/server/JSON.java

```diff
@@ -0,0 +1,208 @@
+
+import codeu.chat.util.Time;
+import codeu.chat.util.Uuid;
+import com.fasterxml.jackson.core.*;
+import com.fasterxml.jackson.databind.*;
+
+/**
+ * This file is to be used to allow the server to get a log of all actions completed
+ * This class saves all inputted commands
+ * It creates a new command so that the user can then access the log
+ * This log can be used by the server to re-load all commands from previous
+ * uses of the server, and to save all users and conversations on the server
+ */
+
+
+public final class JSON {
+
+    public Model createModelForServer()
+    {
+        // This function is to be used by the Server class
+        // If the file, input.json, which holds the transactions
+        // and commands used into the server at a previous time,
+        // exists, this means that the server had been used before.
+        // This will then call readFromFile using that previous
+        // log, allowing the sever to access all previous transactions
+        // If the server hasn't been used before, then there have
+        // been no transactions and therefore there has not yet been a
+        // file created, causing the IOException to be thrown.
+        // Other times when this exception can be thrown
+        // are when the transactions haven't been converted to JSON
+        // properly. In any of these cases, a new Model instance will be
+        // created, essentially starting the server without any memory of
+        // previous transactions/commands.
+        try
+        {
+            // data.json is the predetermined folder in which
+            // the transaction log will go, located in the Server directory
+            // The path may need to be updated based on the way the local project is set up
+            return readFromFile("data.json");
+        }
+        catch (IOException e)
+        {
+            return new Model();
+        }
+    }
+
+    public Model readFromFile (String file) throws IOException
+    {
+        Model model = new Model();
+        JsonFactory jsonF = new JsonFactory();
+        JsonParser jp = null;
+        try
+        {
+            jp = jsonF.createParser(new FileReader(file));
+        }
+        catch(IOException e)
+        {
+            // If the file is invalid or can't be parsed correctly
+            // a version of Model without any transaction logs in
+            // will be used and returned.
+            System.out.println("Invalid file");
+        }
+       if (jp!=null) {
+            // If the file exists, then the file will be parsed
+           // in this set of code
+           jp.nextToken();
+            while (jp.nextToken() != JsonToken.END_OBJECT)
+            // This ensures that we haven't reached the end of the
+            // file because the data is being parsed in such a way
+            // that this END_OBJECT specifically relates to the
+            // very last } in the json file
+            {
+            if(jp.getText()=="users") {
```

Comment:
You should almost never use `==` to compare `String` values. Use the `equals()` function instead. More info here: https://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java

But basically, `==` compares whether two values are the same instance. This might work because of the way `String` values are stored behind the scenes, but it's almost definitely not what you want.

Make sure to fix this in other places where you're using `==` with `String` values too.